cleanrl_utils/submit_exp.py

```diff
@@ -1,97 +1,70 @@
-                   help='the name of the job queue')
-parser.add_argument('--job-definition', type=str, default="cleanrl",
-                   help='the name of the job definition')
+    help='the wandb key. If not provided, the script will try to read from `netrc`')
 parser.add_argument('--num-seed', type=int, default=2,
-                   help='number of random seeds for experiments')
+    help='number of random seeds for experiments')
+
+# experiment submission
+parser.add_argument('--job-queue', type=str, default="m6gd-medium",
+    help='the name of the job queue')
+parser.add_argument('--docker-tag', type=str, default="vwxyzjn/cleanrl:latest",
+    help='the name of the docker tag')
 parser.add_argument('--num-vcpu', type=int, default=1,
-                   help='number of vcpu per experiment')
+    help='number of vcpu per experiment')
 parser.add_argument('--num-memory', type=int, default=2000,
-                   help='number of memory (MB) per experiment')
+    help='number of memory (MB) per experiment')
 parser.add_argument('--num-gpu', type=int, default=0,
-                   help='number of gpu per experiment')
+    help='number of gpu per experiment')
 parser.add_argument('--num-hours', type=float, default=16.0,
-                   help='number of hours allocated experiment')
-parser.add_argument('--auto-resume-attempts', type=int, default=1,
-                   help='the name of the job queue')
-parser.add_argument('--upload-files-baseurl', type=str, default="",
-                   help='the baseurl of your website if you decide to upload files')
-parser.add_argument('--submit-aws', type=lambda x:bool(strtobool(x)), default=False, nargs='?', const=True,
-                    help='if toggled, script will need to be uploaded')
-
+    help='number of hours allocated experiment')
+parser.add_argument('-b', '--build', type=lambda x:bool(strtobool(x)), default=False, nargs='?', const=True,
+    help='if toggled, the script will build a container')
+parser.add_argument('--archs', type=str, default="linux/arm64,linux/amd64",
+    help='the archs to build the docker container for')
+parser.add_argument('--provider', type=str, default="", choices=["aws"],
+    help='the cloud provider of choice (currently only `aws` is supported)')
 args = parser.parse_args()
+# fmt: on
 
-final_str = f'''
-for seed in {{1..{args.num_seed}}}
-do
-    (sleep 0.3 && nohup xvfb-run -a python {args.algo} \\
-    {args.other_args} \\
-    --seed $seed
-    ) >& /dev/null &
-done
-'''
-
-with open(f"{args.exp_script}", "w+") as f:
-    f.write(final_str)
+if args.build:
+    subprocess.run(
+        f"docker buildx build --push --cache-to type=local,mode=max,dest=docker_cache/cleanrl --cache-from type=local,src=docker_cache/cleanrl --platform {args.archs} -t {args.docker_tag} .",
+        shell=True,
+        check=True,
+    )
 
 if not args.wandb_key:
     args.wandb_key = requests.utils.get_netrc_auth("https://api.wandb.ai")[-1]
```

Comment:
Regarding the section about the Wandb configuration when submitting experiments to the cloud, this is the relevant line.
I think the API what is returned by `request.utils.get_netrc_auth` is not as expected.

Reply:
I think the check for the wandb key is not working as expected.

Reply:
Thanks for testing it out. Can you try `wandb login` after you remove the `WANDB_API_KEY` variable?

Reply:
Explicitly logging in with the `wandb login` command and specifying the WANDB Api key seems to make it work as intended.

(Before, the logging-in was done implicitly when queuing up experiments using a simple `runs.sh`, so it was probably not adding the relevant authentication files to the .netrc file).

Reply:
Hmm, let's keep it as it is. We can't handle all the special cases.

Reply:
I concur.